libs/datamodel/core/src/dml/field.rs

```diff
@@ -319,6 +319,14 @@ impl ScalarField {
     pub fn is_optional(&self) -> bool {
         self.arity.is_optional()
     }
+
+    pub fn is_auto_increment(&self) -> bool {
```

Comment:
I think the migration engine already has similar code to check this. Probably it would be best to centralize this logic on the `ValueGenerator` struct and/or `DefaultValue` enum.

Reply:
I found an occurrence in the datamodel converter that was now overshadowed. I'll leave it on the scalar field though, having it on the DefaultValue and Generator would mean I always need to match on these beforehand

Reply:
I think there's an occurence in the migration engine — I'll standardize on the datamodel one if I bump on it.